hack/istio/install-bookinfo-demo.sh

```diff
@@ -399,12 +400,21 @@ fi
 if [ "${TRAFFIC_GENERATOR_ENABLED}" == "true" ]; then
   echo "Installing Traffic Generator"
   if [ "${IS_OPENSHIFT}" == "true" ]; then
-    echo "Determining the route to send traffic to"
+    echo "Determining the route to send traffic to, 1st, trying istio-ingressgateway route in ${INGRESS_NAMESPACE} namespace"
+    # first, try istio-ingressgateway in istio-system
     INGRESS_ROUTE=$(${CLIENT_EXE} get route istio-ingressgateway -o jsonpath='{.spec.host}{"\n"}' -n ${INGRESS_NAMESPACE})
-    while [ -z "${INGRESS_ROUTE}" ]; do
+    if [ -z "${INGRESS_ROUTE}" ]; then
       sleep 1
-      INGRESS_ROUTE=$(${CLIENT_EXE} get route productpage -o jsonpath='{.spec.host}{"\n"}' -n ${NAMESPACE})
-    done
```

Comment:
I'm guessing this was a while loop before to give the ingress route time to be populated? We may need to check more than once before giving up and going onto bookinfo-gateway-istio. Maybe wait for the ingress status to have a `host` for some reasonable timeout?
```
kubectl wait '--for=jsonpath={.status.ingress[].host}' --timeout=10s route istio-ingressgateway -n ${INGRESS_NAMESPACE}
```

Reply:
ah, yea. 
However, to support JSONPath condition without value, some newer kubectl/oc is needed, I am not sure which version is used in our CI

Reply:
Ok maybe then just keep the while loop but exit after ~10 tries?

Reply:
It works on our CI with
```
oc version
Client Version: 4.16.6
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: 4.17.4
Kubernetes Version: v1.30.5
```
so I left it in that way.